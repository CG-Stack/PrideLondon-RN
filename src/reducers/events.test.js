import reducer from "./events";

describe("Events reducer", () => {
  it("initialises with default state", () => {
    // $FlowFixMe
    const state = reducer(undefined, {});

    expect(state).toMatchSnapshot();
  });

  it("sets loading flag for REQUEST_CMS_DATA action", () => {
    const initialState = {
      data: {
        entries: []
      },
      loading: false,
      refreshing: false
    };
    const state = reducer(initialState, { type: "REQUEST_CMS_DATA" });

    expect(state.loading).toBe(true);
    expect(state.refreshing).toBe(false);
  });

  it("sets refreshing flag for REQUEST_UPDATE_CMS_DATA action", () => {
    const initialState = {
      data: {
        entries: []
      },
      loading: false,
      refreshing: false
    };
    const state = reducer(initialState, { type: "REQUEST_UPDATE_CMS_DATA" });

    expect(state.loading).toBe(false);
    expect(state.refreshing).toBe(true);
  });

  it("saves events from RECEIVE_CMS_DATA action", () => {
    const initialState = {
      data: {
        entries: []
      },
      loading: true,
      refreshing: false
    };
    const newCmsData = { entries: [{ id: "1" }] };
    const state = reducer(initialState, {
      type: "RECEIVE_CMS_DATA",
      payload: newCmsData
    });

    expect(state.loading).toBe(false);
    expect(state.refreshing).toBe(false);
    expect(state.data).toBe(newCmsData);
  });
});
