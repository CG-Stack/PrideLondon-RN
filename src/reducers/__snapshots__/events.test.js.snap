// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events reducer initialises with default state 1`] = `
Object {
  "data": Object {
    "assets": Array [],
    "entries": Array [],
  },
  "loading": true,
  "refreshing": false,
}
`;
